### PR TITLE
fix logging in imagegen-config files

### PR DESCRIPTION
**Pull request by neha170**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
fix error logging in imagegen/configuration utilities

###### Change Log  

- fix error logging in imagegen/configuration utilities

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #[49082611](https://microsoft.visualstudio.com/OS/_workitems/edit/49082611)

###### Test Methodology

- Local build

### CODE REVIEW

File: toolkit/tools/imagegen/configuration/networkconfig.go [hunk 1]
```diff
@@ -243,37 +243,32 @@ func createNetworkConfigFile(installChroot *safechroot.Chroot, networkData Netwo
...
 
 	defer func() {
 		// Delete the network file on failure
 		if err != nil {
-			err = os.Remove(networkFilePath)
-			if err != nil {
-				logger.Log.Errorf("Failed to clean up network file (%s). Error: %s", networkFilePath, err)
+			cleanupErr := os.Remove(networkFilePath)
+			if cleanupErr != nil {
+				err = fmt.Errorf("failed to clean up network file (%s):\n%w\n%w", networkFilePath, cleanupErr, err)
```
Comment (cwize1) [inline toolkit/tools/imagegen/configuration/networkconfig.go:271]: `err = fmt.Errorf("failed to clean up network file (%s):\n%w\ncleanup-error:\n%w", networkFilePath, err, cleanupErr)`

If you are going to add cleanup-errors to the error message, then you should clearly separate the original error from the cleanup error.

**Review by cwize1 — CHANGES_REQUESTED** [review]

File: toolkit/tools/imagegen/configuration/networkconfig.go (hunk 1 above)
Comment (neha170) [reply toolkit/tools/imagegen/configuration/networkconfig.go:271]: how about
`err = fmt.Errorf("%w\nfailed to clean up network file (%s):\ncleanup-error:\n%w", err, networkFilePath, cleanupErr)`

Comment (cwize1) [reply toolkit/tools/imagegen/configuration/networkconfig.go:271]: Oh, you're right. But possibly:

```golang
err = fmt.Errorf("%w\ncleanup-error: failed to clean up network file (%s):\n%w", err, networkFilePath, cleanupErr)
```

**Review by cwize1 — CHANGES_REQUESTED** [review]
(See previous comments.)

**Review by cwize1 — APPROVED** [review]

**Pull request merged**